pkg/signatures/factory.go

```diff
@@ -10,10 +10,15 @@ import (
 	"github.com/stackrox/rox/pkg/protoconv"
 	registryTypes "github.com/stackrox/rox/pkg/registries/types"
 	"github.com/stackrox/rox/pkg/retry"
+	"golang.org/x/time/rate"
 )
 
 var (
 	log = logging.LoggerForModule()
+
+	// registryRateLimiter is a rate limiter for parallel calls to the registry. This will avoid reaching out to the
+	// registry too many times leading to 429 errors.
+	registryRateLimiter = rate.NewLimiter(rate.Every(50*time.Millisecond), 1)
```

Comment:
I guess the downside to this solution is that we introduce a global state here. Alternative would be to create a type encapsulating registry rate limiter which would have `FetchImageSignaturesWithRetries` as a receiver function.

Reply:
Hm, I guess what we also could do is to have a rate limiter associated with a `signatures.SignatureFetcher` and wait for the rate limiter within `SignatureFetcher.FetchSignatures`. I think I like this the most, WDYT?

Reply:
I like this option the most as well. Already forgot we have this abstraction, but look - implementing it early paid off :)